src/main/java/org/ow2/proactive/catalog/service/BucketService.java

```diff
@@ -83,6 +90,25 @@ public BucketMetadata createBucket(String name, String owner) throws DataIntegri
         return new BucketMetadata(bucketEntity, 0);
     }
 
+    public BucketMetadata updateOwnerByBucketName(String bucketName, String owner)
+            throws DataIntegrityViolationException {
+        BucketEntity bucketEntity = findBucketByNameAndCheck(bucketName);
+        bucketEntity.setOwner(owner);
+
+        bucketEntity = bucketRepository.save(bucketEntity);
+
+        createRevisionForObjects(bucketName, COMMIT_MESSAGE_UPDATE_BUCKET);
+
+        return new BucketMetadata(bucketEntity, bucketEntity.getCatalogObjects().size());
+    }
+
+    //create a new revision for objects when the bucket owner is updated
+    protected void createRevisionForObjects(String bucketName, String commitMessage) {
+        List<CatalogObjectRevisionEntity> objectsList = catalogObjectService.listCatalogObjectsEntities(Arrays.asList(bucketName));
+
+        objectsList.forEach(obj -> catalogObjectService.createCatalogObjectRevision(obj, commitMessage));
```

Comment:
Does this new revision will appear in the revisions list panel in the Catalog Portal ? Otherwise, what is the purpose to create an object revision ?
Thanks

Reply:
yes, it will appear as the normal new revision. This is the way how to update the key-value metadata values, as they are connected with revisions.